docs/source/custom_task.rst

```diff
@@ -161,34 +161,36 @@ generated ``dataset``.
 
 .. note::
 
-    Why introduce :class:`~flash.data.process.Preprocess` ?
+    Why introduce :class:`~flash.core.data.process.Preprocess` ?
 
-    The :class:`~flash.data.process.Preprocess` object reduces the engineering overhead to make inference on raw data or
+    The :class:`~flash.core.data.process.Preprocess` object reduces the engineering overhead to make inference on raw data or
     to deploy the model in production environnement compared to a traditional
     `Dataset <https://pytorch.org/docs/stable/data.html#torch.utils.data.Dataset>`_.
 
     You can override ``predict_{hook_name}`` hooks or the ``default_predict_transforms`` to handle data processing logic
     specific for inference.
 
-The recommended way to define a custom :class:`~flash.data.process.Preprocess` is as follows:
+The recommended way to define a custom :class:`~flash.core.data.process.Preprocess` is as follows:
 
 - Define an ``__init__`` which accepts transform arguments.
 - Pass these arguments through to ``super().__init__`` and specify the ``data_sources`` and the ``default_data_source``.
-    - ``data_sources`` gives the :class:`~flash.data.data_source.DataSource` objects that work with your :class:`~flash.data.process.Preprocess` as a mapping from data source name to :class:`~flash.data.data_source.DataSource`. The data source name can be any string, but for our purposes we can use ``NUMPY`` from :class:`~flash.data.data_source.DefaultDataSources`.
+    - ``data_sources`` gives the :class:`~flash.core.data.data_source.DataSource` objects that work with your :class:`~flash.core.data.process.Preprocess` as a mapping from data source name to :class:`~flash.core.data.data_source.DataSource`. The data source name can be any string, but for our purposes we can use ``NUMPY`` from :class:`~flash.core.data.data_source.DefaultDataSources`.
     - ``default_data_source`` is the name of the data source to use by default when predicting.
-- Override the ``get_state_dict`` and ``load_state_dict`` methods. These methods are used to save and load your :class:`~flash.data.process.Preprocess` from a checkpoint.
+- Override the ``get_state_dict`` and ``load_state_dict`` methods. These methods are used to save and load your :class:`~flash.core.data.process.Preprocess` from a checkpoint.
 - Override the ``{train,val,test,predict}_default_transforms`` methods to specify the default transforms to use in each stage (these will be used if the transforms passed in the ``__init__`` are ``None``).
-    - Transforms are given as a mapping from hook name to callable transforms. You should use :class:`~flash.data.transforms.ApplyToKeys` to apply each transform only to specific keys in the data dictionary.
+    - Transforms are given as a mapping from hook name to callable transforms. You should use :class:`~flash.core.data.transforms.ApplyToKeys` to apply each transform only to specific keys in the data dictionary.
 
 .. testcode:: custom_task
 
+    from flash import Preprocess
```

Comment:
Imports are defined in part 1, any reason to define them again here? We could just always import locally and remove part 1 to be consistent.